### PR TITLE
Netlink minor fix

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- [netlink: Format and algin netlink.h](https://github.com/apache/nuttx/commit/a940b7206d7936a64b10cb01f5ff44c7f192413a)
- [netlink: Remove the dead code in netlink_close](https://github.com/apache/nuttx/commit/f089f2e3d5213709b821d5fc4468ff77cc9e7a42)

## Impact

code refectory only, no function change.

## Testing

ci

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



The PR you provided **partially meets** the NuttX requirements. 

**Here's why:**

* **Summary:**  While it links to the commits, it lacks a descriptive summary.  It should briefly explain *why* the changes are necessary (e.g., improve code readability, fix a bug, etc.) and *what* was changed (e.g., code formatting in netlink.h, removal of dead code in netlink_close).
* **Impact:** Stating "code refectory only, no function change" is a good start, but it needs to be explicitly stated for *each* impact category whether it's NO or YES. Since this is a refactoring, most likely all will be NO, but each needs to be explicitly marked as such.
* **Testing:** Simply stating "ci" is insufficient. You need to provide:
    * Specifics about your build host(s) and target(s) (OS, CPU architecture, compiler, board configuration, etc.)
    * Actual testing logs from *before* and *after* the changes.  This demonstrates that the code functioned correctly before your changes and continues to function correctly afterward.

**To fully meet the requirements, you need to:**

1. **Expand the Summary:**  Provide a brief, informative description of the changes.
2. **Complete the Impact section:** Explicitly mark each impact category as NO or YES.
3. **Provide detailed testing information:**  Include specific details about your testing environment and actual testing logs from before and after the changes.

File: include/netpacket/netlink.h [hunk 1]
```diff
@@ -811,86 +801,86 @@ enum
 {
   NDUSEROPT_UNSPEC,
   NDUSEROPT_SRCADDR,
-  __NDUSEROPT_MAX
+  NDUSEROPT_MAX
```
Comment (wengzhe) [inline include/netpacket/netlink.h:804]: ditto

File: include/netpacket/netlink.h [hunk 2]
```diff
@@ -725,11 +719,9 @@ enum
   PREFIX_UNSPEC,
   PREFIX_ADDRESS,
   PREFIX_CACHEINFO,
-  __PREFIX_MAX
+  PREFIX_MAX
```
Comment (wengzhe) [inline include/netpacket/netlink.h:722]: Please note that [linux](https://github.com/torvalds/linux/blob/v6.8/include/uapi/linux/rtnetlink.h#L593)'s `PREFIX_MAX` is equal to `PREFIX_CACHEINFO`

File: include/netpacket/netlink.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/netpacket/netlink.h]: Fixed.

File: include/netpacket/netlink.h (hunk 2 above)
Comment (xiaoxiang781216) [reply include/netpacket/netlink.h]: Fixed.

**Review by wengzhe — APPROVED** [review]
LGTM

**Review by jerpelea — APPROVED** [review]
please add a commit message with explanation for this change

**Comment by jerpelea** [comment]
I can not see any explanation in the commit

"netlink: Remove the dead code in netlink_close

Signed-off-by: Xiang Xiao <xiaoxiang@xiaomi.com>"

Best regards
Alin

**Comment by xiaoxiang781216** [comment]
> I can not see any explanation in the commit
> 
> "netlink: Remove the dead code in netlink_close
> 
> Signed-off-by: Xiang Xiao [xiaoxiang@xiaomi.com](mailto:xiaoxiang@xiaomi.com)"
> 
> Best regards Alin

`dead code` is very clear, could you review the change?

**Pull request merged**